source/common/http/alternate_protocols_cache_impl.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include "envoy/common/http/alternate_protocols_cache.h"
+#include "envoy/common/optref.h"
+#include "envoy/common/time.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Http {
+
+// Tracks alternate protocols that can be used to make an HTTP connection to an origin server.
+// See https://tools.ietf.org/html/rfc7838 for HTTP Alternate Services and
+// https://datatracker.ietf.org/doc/html/draft-ietf-dnsop-svcb-https-04 for the
+// "HTTPS" DNS resource record.
+class AlternateProtocolsCacheImpl : public AlternateProtocolsCache {
+public:
+  explicit AlternateProtocolsCacheImpl(TimeSource& time_source);
+  ~AlternateProtocolsCacheImpl() override;
+
+  // Sets the possible alternative protocols which can be used to connect to the
```

Comment:
Now these are overrides the common thing to do is

// AlternateProtocolsCache
block of uncommented override functions

since it's "known" you refer to the API for what it's doing

Reply:
Ah, good point. Done.